full-deps/fulldepsspack/Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM 	   dealii/ubuntu16
+
+MAINTAINER luca.heltai@gmail.com
+
+USER 	   root
+
+# general environment for docker
+ENV        DEBIAN_FRONTEND=noninteractive \
+           SPACK_ROOT=/usr/local \
+           FORCE_UNSAFE_CONFIGURE=1
+
+# install minimal spack dependencies
+RUN        apt-get update \
+           && apt-get install -y --no-install-recommends \
+              autoconf \
```

Comment:
i think some of those are not needed really, `curl` and `git` `gfortran` have to be here, but with the current setup of `packages.yaml` the rest will be built by Spack. One could of course set a couple of those packages as externally provided, but for those small packages it does not take much time to build so i would not bother with this.

Reply:
which ones do you suggest I remove?

Reply:
all except curl, git and gfortran?

Reply:
i would try without: `autoconf`, `python`, `nano`, `environment-modules`.  But we can merge it as-is and if this list can be shorten, then we can do this in another PR.

Reply:
I still need `build-essentials` and python, right?

Reply:
oh, `python` of course.

Reply:
`build-essentials` as well, i think. So it's only `autoconf`, `nano`, `environment-modules`